lib/config/test_configuration.js

```diff
@@ -0,0 +1,285 @@
+        console.log('[get connection to topology]');
+        if (err) return callback(err);
+
+        _db.dropDatabase(function(_err) {
+          console.log('[dropped database]');
+          _db.close();
+          callback();
+        });
+      });
+    }).catch(function(err) {
+      callback(err);
+    });
+};
+
+NativeConfiguration.prototype.newDbInstance = function(dbOptions, serverOptions) {
+  serverOptions = serverOptions || {};
+  // Override implementation
+  if (this.options.newDbInstance) {
+    return this.options.newDbInstance(dbOptions, serverOptions);
+  }
+
+  // Set up the options
+  var keys = Object.keys(this.options);
+  if (keys.indexOf('sslOnNormalPorts') !== -1) serverOptions.ssl = true;
+
+  // Fall back
+  var dbHost = serverOptions && serverOptions.host || 'localhost';
+  var dbPort = serverOptions && serverOptions.port || this.options.port || 27017;
+
+  // Default topology
+  var DbTopology = this.mongo.Server;
+  // If we have a specific topology
+  if (this.options.topology) {
+    DbTopology = this.options.topology;
+  }
+
+  // Return a new db instance
+  return new this.mongo.Db(this.db, new DbTopology(dbHost, dbPort, serverOptions), dbOptions);
+};
+
+NativeConfiguration.prototype.newDbInstanceWithDomainSocket = function(dbOptions, serverOptions) {
+  // Override implementation
+  if (this.options.newDbInstanceWithDomainSocket) return this.options.newDbInstanceWithDomainSocket(dbOptions, serverOptions);
+
+  // Default topology
+  var DbTopology = this.mongo.Server;
+  // If we have a specific topology
+  if (this.options.topology) {
+    DbTopology = this.options.topology;
+  }
+
+  // Fall back
+  var dbHost = serverOptions && serverOptions.host || '/tmp/mongodb-27017.sock';
+
+  // Set up the options
+  var keys = Object.keys(this.options);
+  if (keys.indexOf('sslOnNormalPorts') !== -1) serverOptions.ssl = true;
+  // If we explicitly testing undefined port behavior
+  if (serverOptions && serverOptions.port === 'undefined') {
+    return new this.mongo.Db(this.db, new DbTopology(dbHost, undefined, serverOptions), dbOptions);
+  }
+
+  // Normal socket connection
+  return new this.mongo.Db(this.db, new DbTopology(dbHost, serverOptions), dbOptions);
+};
+
+NativeConfiguration.prototype.url = function(username, password) {
+  var url = this.options.url || 'mongodb://%slocalhost:27017/' + this.db;
+  // Fall back
+  var auth = '';
+  if (username && password) {
+    auth = f('%s:%s@', username, password);
+  }
+
+  return f(url, auth);
+};
+
+NativeConfiguration.prototype.writeConcern = function() {
+  return clone(this.options.writeConcern || {w: 1});
+};
+
+NativeConfiguration.prototype.writeConcernMax = function() {
+  return clone(this.options.writeConcernMax || {w: 1});
+};
+
+var clone = function(obj) {
+  var copy = {};
+  for (var name in obj) copy[name] = obj[name];
+  return copy;
+};
+
+module.exports = {
```

Comment:
I would probably export `ConfigurationBase` as well, just in the unlikely case that we actually want to make some other configuration